src/foam/nanos/google/api/sheets/import/GoogleSheetsDataImportServiceImpl.js

```diff
@@ -0,0 +1,416 @@
+    'java.math.BigDecimal',
+    'java.text.ParseException',
+    'java.text.SimpleDateFormat',
+    'java.time.LocalDateTime',
+    'java.time.format.DateTimeFormatter',
+    'java.util.ArrayList',
+    'java.util.Date',
+    'java.util.List',
+    'java.util.Locale',
+    'java.util.regex.Matcher',
+    'java.util.regex.Pattern',
+
+    'static foam.mlang.MLang.AND',
+    'static foam.mlang.MLang.CONTAINS_IC',
+    'static foam.mlang.MLang.EQ'
+  ],
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        //move all this to helper class?
+        cls.extras.push(foam.java.Code.create({
+          data: `
+            public static Pattern digitAppearenceRegex = Pattern.compile("(\\\\d){1}");
+            public static Pattern numbersRegex = Pattern.compile("\\\\d+(\\\\.\\\\d{1,2})?");
+            public static Pattern alphabeticalCharsRegex = Pattern.compile("[a-zA-Z]{1,}");
+          `
+        }));
+      }
+    }
+  ],
+  methods: [
+    {
+      name: 'getColumns',
+      javaType: 'String[]',
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'importConfig',
+          type: 'foam.nanos.google.api.sheets.GoogleSheetsImportConfig'
+        }
+      ],
+      javaCode:`
+        GoogleSheetsApiService googleSheetsAPIEnabler = (GoogleSheetsApiService)x.get("googleSheetsDataExport");
+        ValueRange values;
+        try {
+          values = googleSheetsAPIEnabler.getFormatedValues(x, importConfig.getGoogleSpreadsheetId(), importConfig.getGoogleSheetId());//sb.toString()
+          List<Object> firstRow = values.getValues().get(0);
+          String[] columnNames = new String[firstRow.size()];
+          for ( int i = 0 ; i < firstRow.size() ; i++ ) {
+            columnNames[i] = String.valueOf(firstRow.get(i));
+          }
+          return columnNames;
+        } catch ( Throwable t ) {
+          Logger logger = (Logger) x.get("logger");
+          if ( logger == null ) {
+            logger = new StdoutLogger();
+          }
+          logger.error(t);
+          return null;
+        }
+      `
+    },
+    {
+      name: 'importData',
+      type: 'foam.nanos.google.api.sheets.ImportDataMessage',
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'importConfig',
+          type: 'foam.nanos.google.api.sheets.GoogleSheetsImportConfig'
+        }
+      ],
+      javaCode: `
+        GoogleSheetsApiService googleSheetsAPIEnabler = (GoogleSheetsApiService)x.get("googleSheetsDataExport");
+        ValueRange values;
+        ImportDataMessage result = new ImportDataMessage();
+        try {
+          values = googleSheetsAPIEnabler.getFormatedValues(x, importConfig.getGoogleSpreadsheetId(), importConfig.getGoogleSheetId());
+          
+          importConfig.setCellsRange(values.getRange().split("!")[1]);
+
+          List<List<Object>> data = values.getValues();
```

Comment:
suggestion: 
if there was a problem with adding records we still might want to update user's google sheet with ids or status
so we might need extra logic for adding status that might be too much thought